packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -27,57 +29,74 @@ export default function ClassifierWrapper({
   authClient = auth,
   appLoadingState = asyncStates.initialized,
   cachePanoptesData = false,
-  collections,
+  collections = null,
   mode,
-  onAddToCollection = () => true,
-  onSubjectReset = () => true,
-  project,
-  recents,
-  router,
+  onAddToCollection = DEFAULT_HANDLER,
+  onSubjectReset = DEFAULT_HANDLER,
+  project = null,
+  recents = null,
+  router = null,
   showTutorial = false,
   subjectID,
   subjectSetID,
-  user,
+  user = null,
   userID,
   workflowID,
   yourStats
 }) {
+  const [classifierSubjectID, setClassifierSubjectID] = useState(subjectID)
   const { adminMode } = useAdminMode()
   const nextRouter = useRouter()
   router = router || nextRouter
   const locale = router?.locale
   const ownerSlug = router?.query.owner
   const projectSlug = router?.query.project
 
+  const incrementStats = yourStats?.increment
+  const addRecents = recents?.add
   const onCompleteClassification = useCallback((classification, subject) => {
     const finishedSubject = subject.already_seen || subject.retired
     if (!finishedSubject) {
-      yourStats.increment()
+      incrementStats()
     }
-    recents.add({
+    addRecents({
       favorite: subject.favorite,
       subjectId: subject.id,
       locations: subject.locations
     })
-  }, [recents?.add, yourStats?.increment])
+  }, [addRecents, incrementStats])
 
-  const onSubjectChange = useCallback((subject) => {
-    const baseURL = `/${ownerSlug}/${projectSlug}/classify`
-    if (subjectID && subject.id !== subjectID) {
-      const newSubjectRoute = `${baseURL}/workflow/${workflowID}/subject-set/${subjectSetID}/subject/${subject.id}`
+  /*
+    If the page URL contains a subject ID, update that ID when the classification subject changes.
+  */
+  const replaceRoute = router.replace
+  let baseURL = `/${ownerSlug}/${projectSlug}/classify/workflow/${workflowID}`
+  if (subjectSetID) {
+    baseURL = `${baseURL}/subject-set/${subjectSetID}`
+  }
+  const subjectRouteChanged = router?.query.subjectID && classifierSubjectID !== router.query.subjectID
+
+  useEffect(function updatePageURL() {
+    if (subjectRouteChanged) {
+      const newSubjectRoute = `${baseURL}/subject/${classifierSubjectID}`
       const href = addQueryParams(newSubjectRoute)
       const as = href
-      router.replace(href, as, { shallow: true })
+      replaceRoute(href, as, { shallow: true })
     }
-  }, [ownerSlug, projectSlug, router?.replace, subjectID, subjectSetID, workflowID])
+  }, [baseURL, replaceRoute, subjectRouteChanged])
 
+  /*
+    Track the current classification subject, when it changes inside the classifier.
+  */
+  const onSubjectChange = useCallback((subject) => {
+    setClassifierSubjectID(subject?.id)
+  }, [setClassifierSubjectID])
```

Comment:
Do `setState` calls have to be declared as dependencies in hooks that set state?

Reply:
Apparently not:
>React guarantees that `setState` function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the `useEffect` or `useCallback` dependency list.

https://legacy.reactjs.org/docs/hooks-reference.html#usestate